iguana/iguana_notary.c

```diff
@@ -300,6 +297,59 @@ void iguana_dPoWupdate(struct supernet_info *myinfo,struct dpow_info *dp)
     } else printf("iguana_dPoWupdate missing src.(%s) %p or dest.(%s) %p\n",dp->symbol,src,dp->dest,dest);
 }
 
+int32_t iguana_BN_dPoWupdate(struct supernet_info *myinfo,struct dpow_info *dp)
+{
+    /*
+    Used with the following. Just change KMD to the coin name being dpowd and put this in the conf file.
+    blocknotify=curl -s --url "http://127.0.0.1:7776" --data "{\"agent\":\"dpow\",\"method\":\"updatechaintip\",\"blockhash\":\"%s\",\"symbol\":\"KMD\"}"
+    */
+
+    int32_t height,flag=0; uint32_t blocktime; bits256 blockhash,merkleroot; struct iguana_info *src,*dest;
+    src = iguana_coinfind(dp->symbol);
+    dest = iguana_coinfind(dp->dest);
+    if ( src != 0 && dest != 0 )
+    {
+        dp->numdesttx = sizeof(dp->desttx)/sizeof(*dp->desttx);
+        if ( (height= dpow_getchaintip(myinfo,&merkleroot,&blockhash,&blocktime,dp->desttx,&dp->numdesttx,dest)) != dp->destchaintip.blockhash.height && height >= 0 )
+        {
+            char str[65];
+            //printf("[%s] %s height.%d vs last.%d\n",dp->dest,bits256_str(str,blockhash),height,dp->destchaintip.blockhash.height);
+            if ( height <= dp->destchaintip.blockhash.height )
+            {
+                printf("iguana_BN_dPoWupdate dest.%s reorg detected %d vs %d\n",dp->dest,height,dp->destchaintip.blockhash.height);
+                if ( height == dp->destchaintip.blockhash.height && bits256_cmp(blockhash,dp->destchaintip.blockhash.hash) != 0 )
```

Comment:
Correct me if I'm wrong, but as I read it:

line 313 sets height to the result of the `dpow_getchaintip`
The conditional for 313 is if height _isn't_ equal to dp->destchaintip.blockhash.height
How could height ever be equal to dp->destchaintip.blockhash.height within line 320?